jgrapht-core/src/main/java/org/jgrapht/traverse/LexicographicalBfsIterator.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * (C) Copyright 2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.traverse;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphTests;
+import org.jgrapht.Graphs;
+
+import java.util.*;
+
+/**
+ * A lexicographical breadth-first iterator for an undirected graph.
+ * <p>
+ * Every vertex has an implicit label (they aren't used explicitly in order to reduce time and
+ * memory complexity). When some vertex is returned by this iterator, its index is the number of
+ * vertices in this graph minus number of already returned vertices. For a given vertex v its label
+ * is a concatenation of indices of already returned vertices, that were also its neighbours, with
+ * some separator between them. For example, 7#4#3 is a valid vertex label.
+ * <p>
+ * Iterator chooses vertex with lexicographically largest label and returns it. It breaks ties
+ * arbitrarily. For more information on lexicographical BFS see the following papers: <a
+ * href="https://pdfs.semanticscholar.org/d4b5/a492f781f23a30773841ec79c46d2ec2eb9c.pdf">
+ * <i>Lexicographic Breadth First Search – A Survey</i></a> and <a
+ * href="http://www.cse.iitd.ac.in/~naveen/courses/CSL851/uwaterloo.pdf"><i>CS 762: Graph-theoretic
+ * algorithms. Lecture notes of a graduate course. University of Waterloo</i></a>.
+ * <p>
+ * For this iterator to work correctly the graph must not be modified during iteration. Currently
+ * there are no means to ensure that, nor to fail-fast. The results of such modifications are
+ * undefined.
+ *
+ * @param <V> the graph vertex type.
+ * @param <E> the graph edge type.
+ * @author Timofey Chudakov
+ * @since 1.8
+ */
+public class LexicographicalBfsIterator<V, E> extends AbstractGraphIterator<V, E> {
```

Comment:
You're extending AbstractGraphIterator, but you're not actually implementing any of the GraphIterator interface.  So you should either make addTraversalListener throw unsupported, or else just extend Iterator instead of GraphIterator.